Add support for C++20

Hi,

The project doesn't compile if the language standard is set to ISO C++20 Standard (/std:c++20):

```
2>------ Build started: Project: stltest, Configuration: Debug x64 ------
2>Building 'stltest' with toolset 'WindowsKernelModeDriver10.0' and the 'Desktop' target platform.
2>deque_tests.cpp
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(469,66): error C2027: use of undefined type 'std::_Replace_first_parameter<_Other,_Ty>'
2>        with
2>        [
2>            _Other=int,
2>            _Ty=jxy::details::allocator<int,PagedPool,809976148>
2>        ]
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(469): message : see declaration of 'std::_Replace_first_parameter<_Other,_Ty>'
2>        with
2>        [
2>            _Other=int,
2>            _Ty=jxy::details::allocator<int,PagedPool,809976148>
2>        ]
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(582): message : see reference to class template instantiation 'std::_Get_rebind_type<_Alloc,_Ty,void>' being compiled
2>        with
2>        [
2>            _Alloc=jxy::details::allocator<int,PagedPool,809976148>,
2>            _Ty=int
2>        ]
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(756): message : see reference to alias template instantiation 'std::_Normal_allocator_traits<_Alloc>::rebind_alloc<_Ty>' being compiled
2>        with
2>        [
2>            _Alloc=jxy::details::allocator<int,PagedPool,809976148>,
2>            _Ty=int
2>        ]
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(582): message : see reference to alias template instantiation 'std::_Rebind_alloc_t<jxy::details::allocator<T,PagedPool,809976148>,_Ty>' being compiled
2>        with
2>        [
2>            T=int,
2>            _Ty=int
2>        ]
2>E:\Files\Code\C++\keval\kevald\stlkrn\stltest\deque_tests.cpp(16): message : see reference to class template instantiation 'std::deque<int,jxy::details::allocator<T,PagedPool,809976148>>' being compiled
2>        with
2>        [
2>            T=int
2>        ]
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(469,60): error C2061: syntax error: identifier 'type'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(573,69): error C2794: 'type': is not a member of any direct or indirect base class of 'std::_Get_rebind_type<_Alloc,_Ty,void>'
2>        with
2>        [
2>            _Alloc=jxy::details::allocator<int,PagedPool,809976148>,
2>            _Ty=int
2>        ]
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(582,53): error C2938: 'std::_Normal_allocator_traits<_Alloc>::rebind_alloc' : Failed to specialize alias template
2>        with
2>        [
2>            _Alloc=jxy::details::allocator<int,PagedPool,809976148>
2>        ]
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(582,53): error C2938: 'std::_Rebind_alloc_t' : Failed to specialize alias template
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(584,76): error C2955: 'std::allocator_traits': use of class template requires template argument list
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(737): message : see declaration of 'std::allocator_traits'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(591,55): error C2955: 'std::allocator_traits': use of class template requires template argument list
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(737): message : see declaration of 'std::allocator_traits'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(591,89): error C2955: 'std::allocator_traits': use of class template requires template argument list
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(737): message : see declaration of 'std::allocator_traits'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(592,36): error C2955: 'std::allocator_traits': use of class template requires template argument list
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(737): message : see declaration of 'std::allocator_traits'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(592,68): error C2955: 'std::allocator_traits': use of class template requires template argument list
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(737): message : see declaration of 'std::allocator_traits'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(592,101): error C2993: 'unknown-type': is not a valid type for non-type template parameter '_Test'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(595,58): error C2955: 'std::_Deque_val': use of class template requires template argument list
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(540): message : see declaration of 'std::_Deque_val'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(600,52): error C2955: 'std::allocator_traits': use of class template requires template argument list
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(737): message : see declaration of 'std::allocator_traits'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(600,52): error C2039: 'size_type': is not a member of 'std::allocator_traits<_Alloc>'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(600,1): error C2061: syntax error: identifier 'size_type'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(601,52): error C2955: 'std::allocator_traits': use of class template requires template argument list
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(737): message : see declaration of 'std::allocator_traits'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(601,52): error C2039: 'difference_type': is not a member of 'std::allocator_traits<_Alloc>'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(601,1): error C2061: syntax error: identifier 'difference_type'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(602,52): error C2955: 'std::allocator_traits': use of class template requires template argument list
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(737): message : see declaration of 'std::allocator_traits'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(602,52): error C2039: 'pointer': is not a member of 'std::allocator_traits<_Alloc>'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(602,1): error C2061: syntax error: identifier 'pointer'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(603,52): error C2955: 'std::allocator_traits': use of class template requires template argument list
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(737): message : see declaration of 'std::allocator_traits'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(603,52): error C2039: 'const_pointer': is not a member of 'std::allocator_traits<_Alloc>'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(603,1): error C2061: syntax error: identifier 'const_pointer'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(607,55): error C3203: '_Deque_val': unspecialized class template can't be used as a template argument for template parameter '_Mydeque', expected a real type
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(608,61): error C3203: '_Deque_val': unspecialized class template can't be used as a template argument for template parameter '_Mydeque', expected a real type
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(609,65): error C3203: '_Deque_val': unspecialized class template can't be used as a template argument for template parameter '_Mydeque', expected a real type
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(610,71): error C3203: '_Deque_val': unspecialized class template can't be used as a template argument for template parameter '_Mydeque', expected a real type
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(614,22): error C2131: expression did not evaluate to a constant
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(595,58): message : failure was caused by non-constant arguments or reference to a non-constant symbol
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(595,58): message : see usage of 'std::_Deque_val<_Val_types>::_Block_size'
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\deque(624,39): error C2061: syntax error: identifier 'size_type'

[...]
```

Help would be greatly appreciated 😺

--- a/include/jxy/memory.hpp
+++ b/include/jxy/memory.hpp
@@ -89,6 +89,12 @@ public:
         return memory;
     }
 
+    template <typename Other>
+    struct rebind
+    {
+        using other = allocator<Other, t_PoolType, t_PoolTag>;
+    };
+
 #if _HAS_DEPRECATED_ALLOCATOR_MEMBERS
     _CXX17_DEPRECATE_OLD_ALLOCATOR_MEMBERS typedef T* pointer;
     _CXX17_DEPRECATE_OLD_ALLOCATOR_MEMBERS typedef const T* const_pointer;
@@ -97,12 +103,6 @@ public:
     _CXX17_DEPRECATE_OLD_ALLOCATOR_MEMBERS typedef const T& const_reference;
 
     using is_always_equal _CXX17_DEPRECATE_OLD_ALLOCATOR_MEMBERS = std::true_type;
-
-    template <typename Other>
-    struct _CXX17_DEPRECATE_OLD_ALLOCATOR_MEMBERS rebind 
-    {
-        using other = allocator<Other, t_PoolType, t_PoolTag>;
-    };
 
     _CXX17_DEPRECATE_OLD_ALLOCATOR_MEMBERS 
     _NODISCARD 

--- a/include/jxy/memory.hpp
+++ b/include/jxy/memory.hpp
@@ -89,6 +89,18 @@ public:
         return memory;
     }
 
+    //
+    // jxy::allocator implements rebind despite it being deprecated (cpp17) and
+    // removed (cpp20) on std::allocator.
+    // Without this, std::_Replace_first_parameter<_Other,_Ty> trips up on replacing
+    // the type at rebind_alloc on allocator_traits.
+    //
+    // Per the allocator concepts - implementation of rebind is optional and seemingly
+    // required for us to avoid the template parameter replacement on the custom
+    // allocator:
+    // https://en.cppreference.com/w/cpp/named_req/Allocator#cite_note-2
+    //
+
     template <typename Other>
     struct rebind
     {

